src/bundle/Resources/public/js/scripts/core/storage.js

```diff
@@ -0,0 +1,57 @@
+(function (global, doc, ibexa) {
+    class IbexaStorage {
+        constructor(config) {
+            const keySuffix = config.isUserAware ? `:${ibexa.helpers.user.getId()}` : '';
+
+            this.key = `${config.key}${keySuffix}`;
+            this.eventName = config.eventName;
+        }
+
+        stringfyData(data) {
+            try {
+                return JSON.stringify(data);
+            } catch (error) {
+                console.warn('Error stringifying data', error);
+            }
+        }
+
+        parseData(data) {
+            try {
+                return JSON.parse(data);
+            } catch (error) {
+                return null;
+            }
+        }
+
+        setItem(data) {
+            const stringifiedData = this.stringfyData(data);
+
+            window.localStorage.setItem(this.key, stringifiedData);
+
+            this.fireStorageChangeEvent(stringifiedData);
+        }
+
+        getItem() {
+            return this.parseData(window.localStorage.getItem(this.key));
+        }
+
+        fireStorageChangeEvent(data) {
+            if (this.eventName) {
+                const storageChangeEvent = new CustomEvent(this.eventName, {
+                    cancelable: true,
+                    detail: { content: this.parseData(data) },
```

Comment:
what would you say on sending key as well?

Reply:
I don't see a use case where the listener would not know the key if it is attached to a specific eventName.